openwebstart/src/main/java/com/openwebstart/launcher/OwsJvmLauncher.java

```diff
@@ -148,25 +145,18 @@ private void launchExternal(
     private List<String> extractVmArgs(final JNLPFile jnlpFile) {
         if (jnlpFile.getSecurity().getApplicationPermissionLevel() == ApplicationPermissionLevel.ALL) {
             final List<String> result = new ArrayList<>();
-
             final Map<String, String> properties = jnlpFile.getResources().getPropertiesMap();
-            addVmArg(result, properties, HTTP_AGENT);
-            addVmArg(result, properties, HTTP_MAX_REDIRECTS);
-            addVmArg(result, properties, HTTP_AUTH_DIGEST_VALIDATEPROXY);
-            addVmArg(result, properties, HTTP_AUTH_DIGEST_VALIDATESERVER);
-            addVmArg(result, properties, AWT_DISABLE_MIXING);
+            properties.keySet().forEach(property -> {
+                if (JvmUtils.isValidSecureProperty(property)) {
+                    result.add(String.format("-D%s=%s", property, properties.get(property)));
```

Comment:
will it cause problems if the value contains a space??

Reply:
<property name="jnlp.abc" value="d e f"/> comes out as -Djnlp.abc=d e f. So it seems the blanks are preserved!